File: tests/benchmark/corpus/python/safe/safe_type_check.py
import sqlite3
from flask import request

def get_user():
    user_id = request.args.get('id')
    if not isinstance(user_id, int):
        return "bad input"
    conn = sqlite3.connect('app.db')
    cursor = conn.cursor()
    cursor.execute("SELECT * FROM users WHERE id = " + str(user_id))
    result = cursor.fetchone()
    cursor.close()
    conn.close()
    return result

File: tests/benchmark/corpus/go/safe/safe_type_check.go
package main

import (
	"database/sql"
	"net/http"
	"strconv"
)

func handler(w http.ResponseWriter, r *http.Request) {
	input := r.URL.Query().Get("id")
	_, err := strconv.Atoi(input)
	if err != nil {
		http.Error(w, "bad input", 400)
		return
	}
	db, _ := sql.Open("sqlite3", "app.db")
	defer db.Close()
	db.Query("SELECT * FROM users WHERE id = " + input)
}
